packages/webdriver/src/constants.js

```diff
@@ -90,3 +90,21 @@ export const DEFAULTS = {
         type: 'string'
     }
 }
+
+export const SESSION_DEFAULTS = Object.assign({}, DEFAULTS, {
+    // Capabilities are optional when attaching session,
+    // due to the fact drivers not always return capabilities
+    // in the initial (JsonWireProtocol) session response.
```

Comment:
In which case does this apply? Drivers should always return the session capabilities.

Reply:
In this case I'm referring to chromedriver, which has "capabilities" stored as `value`.
https://github.com/webdriverio/webdriverio/blob/684551612eb0ff5b97f8a861d53e91e683bbe987/packages/webdriver/tests/__fixtures__/chromedriver.response.json#L3-L4

Whereas other drivers (e.g. geckodriver) return both `sessionId` and `capabilities` in the same object.
https://github.com/webdriverio/webdriverio/blob/684551612eb0ff5b97f8a861d53e91e683bbe987/packages/webdriver/tests/__fixtures__/geckodriver.response.json#L2-L4

So an example snippet for attaching a session would be as followed:
```js
const request = require('request')

async function initSession(browserName) {
    const capabilities = {
        browserName: browserName
    }
    const postData = {
        capabilities: {
            alwaysMatch: capabilities,
            firstMatch: [{}]
        },
        desiredCapabilities: capabilities
    };

    return new Promise(function (resolve, reject) {
        request({
            url: 'http://localhost:4444/wd/hub/session',
            method: 'POST',
            json: postData
        }, function (error, res, body) {
            if (!error && res.statusCode == 200) {
                resolve(body);
            } else {
                reject(error);
            }
        });
    });
}


(async () => {
    const response = await initSession('firefox')
    const session = await attach(response.value)
    console.log(await session.getUrl())
})().catch((e) => console.log(e))
```

Reply:
Or in the case of internally calling the code, which simply does not have `capabilities` available (on top level).
For example when using the `MultiRemote` class which is a wrapper for multiple instances with different capabilities.

https://github.com/webdriverio/webdriverio/blob/5750a9ba4050ddf9a3b310734700d606782432e9/packages/webdriverio/src/index.js#L59-L62